pkg/plugin/plugin.go

```diff
@@ -20,53 +24,58 @@ type TLSFiles struct {
 	CLIENT    string `json:"client"`
 }
 
-func readTlsFile() (TLSFiles, error){
+func readTlsFile() (TLSFiles, error) {
 	home, err := os.UserHomeDir()
 	if err != nil {
-	  return TLSFiles{}, err
+		return TLSFiles{}, err
 	}
 	clientKey, err := os.ReadFile(home + "/.lightning/testnet/client-key.pem")
 	ca, err := os.ReadFile(home + ".lightning/testnet/ca.pem")
 	client, err := os.ReadFile(home + ".lightning/testnet/client.pem")
-	if err != nil{
-	  return TLSFiles{}, err
+	if err != nil {
+		return TLSFiles{}, err
 	}
 	files := TLSFiles{CA: string(ca), CLIENTKEY: string(clientKey), CLIENT: string(client)}
 	return files, nil
 }
 
-func startServer() {
-	r := gin.Default()
-	r.GET("/ping", func(c *gin.Context) {
+type StartServer[T PluginState] struct{}
+
+func (instance *StartServer[T]) Call(plugin *plugin.Plugin[PluginState], request map[string]string) (map[string]string, error) {
+	if plugin.State.Server != nil {
+		return map[string]string{"message": "Server is already running"}, nil
```

Comment:
```suggestion
		return nil, fmt.Error("Server is already running")
```

This is not supported maybe from the cln4go library bug we need to support it, we return the error and the library will return in the correct way the error to core lightning.